Integrate Eclipse Epicyro

arjantijms opened the pull request:
This is the GlassFish code for Jakarta Authentication, extracted to a separate project and refactored.

arjantijms committed:
Integrate Eleos

This is the GlassFish code for Jakarta Authentication, extracted to a
separate project and refactored.

Eleos currently lives in an OmniFish repo, but should be transfered to
Eclipse soon.

See https://projects.eclipse.org/projects/ee4j.glassfish/reviews/eclipse-eleos-creation-review

arjantijms committed:
Merge branch 'master' into eleos

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Use non-snapshot version of Eleos

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Add eleos dependency to embedded

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

dmatej commented on appserver/security/appclient.security/src/main/java/com/sun/enterprise/security/appclient/AppClientSecurityInfoImpl.java:
```diff
@@ -56,7 +50,10 @@
 public class AppClientSecurityInfoImpl implements AppClientSecurityInfo {
 
-    private static final Logger LOG = System.getLogger(AppClientSecurityInfoImpl.class.getName());
+    private static Logger _logger=null;
```
I think here you have to remove your changes.

arjantijms replied:
I did, thanks!

arjantijms committed:
Revert loggers

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Fix compilation errors

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
About logger number after removal of Jaspic provider module

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

OndroMih commented on appserver/security/core-ee/src/main/java/com/sun/enterprise/security/ee/JavaEESecurityLifecycle.java:
```diff
@@ -40,36 +41,39 @@ public class JavaEESecurityLifecycle implements ContainerSecurityLifecycle, Post
     private static final Logger LOG = LogDomains.getLogger(JavaEESecurityLifecycle.class, LogDomains.SECURITY_LOGGER, false);
 
+    @Override
+    public void postConstruct() {
```
I agree. I think it should either be in postConstruct and remove `implements ContainerSecurityLifecycle`, or without postConstruct and executed from the container lifecycle.

OndroMih commented on appserver/security/core-ee/src/main/java/com/sun/enterprise/security/jmac/ConfigDomainParser.java:
```diff
@@ -92,34 +106,24 @@ private void processServerConfig(SecurityService service, Map newConfig) throws
                 if (provList != null) {
 
-                    for (ProviderConfig provider : provList) {
+                    Iterator<ProviderConfig> pit = provList.iterator();
```
Looks like a revert of recent improvements in master branch. For loop is better than Iterator.

OndroMih commented on appserver/security/core-ee/src/main/java/com/sun/enterprise/security/jmac/ConfigDomainParser.java:
```diff
@@ -41,21 +33,34 @@
 public class ConfigDomainParser implements ConfigParser {
 
-    private static final Logger _logger = LogDomains.getLogger(ConfigDomainParser.class, LogDomains.SECURITY_LOGGER);
+    private static Logger _logger = null;
```
Looks like a revert of recent improvements in master branch. it's better to initialize the field directly and not in a separate static block.

OndroMih commented on appserver/security/core-ee/src/main/java/com/sun/enterprise/security/jmac/ConfigDomainParser.java:
```diff
@@ -41,21 +33,34 @@
     // configuration info
-    private final Map configMap = new HashMap();
-    private final Set<String> layersWithDefault = new HashSet<>();
+    private Map<String, AuthModulesLayerConfig>  configMap = new HashMap<>();
+    private Set<String> layersWithDefault = new HashSet<>();
```
These 2 can remain final I think

OndroMih commented on appserver/security/core-ee/src/main/java/com/sun/enterprise/security/jmac/MessageSecurityConfigEventListenerImpl.java:
```diff
@@ -77,7 +76,7 @@ public <T extends ConfigBeanProxy> NotProcessed handleDelete(T instance) {
         LOG.log(Level.FINE, "MessageSecurityConfigEventListenerImpl - handleCreate called");
         if (instance instanceof MessageSecurityConfig) {
-            GFServerConfigProvider.loadConfigContext(service);
+            // GFServerConfigProvider.loadConfigContext(service);
```
What about this? Remove or uncomment? Don't leave commented code in the codebase.

OndroMih commented on appserver/security/core-ee/src/main/java/com/sun/enterprise/security/jmac/callback/ContainerCallbackHandler.java:
```diff
@@ -72,10 +72,8 @@ public String getRealmName() {
             }
         };
-        ((BaseContainerCallbackHandler) handler).setHandlerContext(handlerContext);
+        // TODO
```
Don't forget this ;-)

OndroMih commented on appserver/security/webintegration/pom.xml:
```diff
@@ -118,6 +118,10 @@
             <artifactId>gmbal</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.omnifaces</groupId>
```
After Eclipse Eleos publishes an artifact, should be changed to Eclipse artifact.

OndroMih commented on appserver/security/webservices.security/pom.xml:
```diff
@@ -140,5 +140,10 @@
             <artifactId>jakarta.authorization-api</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.omnifaces</groupId>
```
After Eclipse Eleos publishes an artifact, should be changed to Eclipse artifact.

OndroMih commented on appserver/security/webservices.security/src/main/java/com/sun/enterprise/security/jmac/provider/config/PipeHelper.java:
```diff
@@ -367,7 +368,8 @@ public void disable() {
 
 
-    @Override
+    // TODO
```
Don't forget this ;-)

OndroMih commented on nucleus/core/bootstrap/src/test/java/com/sun/enterprise/glassfish/bootstrap/MainHelperTest.java:
```diff
@@ -96,7 +96,7 @@ void parseAsEnvTest() throws Exception {
 
 
-    @Test
+    // @Test
```
Don't forget this ;-)
Or remove the test if it can't be enabled.

arjantijms committed:
Update devtests security with option to run single test

Also apply formatting and debug option for SOAP tests.

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Continued integration of Eleos into GlassFish

Focussed on SOAP and app client integration.

Specifically added defaults for factories created statically without
arguments.

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Merge branch 'master' into eleos

arjantijms committed:
Small amount of cleanup

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Added refresh of factory, which effectively reads the config again

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Merge branch 'master' into eleos

dmatej commented on appserver/security/appclient.security/src/main/java/com/sun/enterprise/security/appclient/AppClientSecurityInfoImpl.java:
```diff
@@ -46,8 +33,21 @@
 import org.glassfish.appclient.client.acc.config.TargetServer;
 import org.glassfish.enterprise.iiop.api.IIOPSSLUtil;
 import org.jvnet.hk2.annotations.Service;
+import org.omnifaces.eleos.config.factory.file.AuthConfigFileFactory;
```
alphabetic order in imports

arjantijms committed:
Integrate soap policies and key callbacks

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Clean tests a little

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Some cleanup

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Fix logging

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

dmatej commented on appserver/security/appclient.security/src/main/java/com/sun/enterprise/security/appclient/AppClientSecurityInfoImpl.java:
```diff
@@ -101,59 +98,45 @@ public void initializeSecurity(
             UsernamePasswordStore.set(username, password);
         }
 
-        //why am i setting both?.
```
Don't comments without a value, please - even for commenting comments without a value ... maybe you could add FIXME ;-)

arjantijms replied:
Commenting on comments without a value. Sometimes there's a little bit of humour in the misery. I'm not sure if there is anything to be fixed though, maybe only the comment itself needs fixing.

arjantijms committed:
Some cleaning. Removed unused parameter, fixed class ctor, etc

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Merge branch 'master' into eleos

arjantijms committed:
Merge branch 'master' into eleos

arjantijms committed:
Refactoring and Eleos integration for connectors

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Update to Eleos 3.0.4

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Merge branch 'master' into eleos

arjantijms committed:
Merge branch 'master' into eleos

arjantijms committed:
Merge branch 'master' into eleos

arjantijms committed:
Merge branch 'master' into eleos

arjantijms committed:
Merge branch 'master' into eleos

arjantijms committed:
Replace Eleos with Eclipse Epicyro

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Use M1 for Epicyro

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Property for epicyro.version

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Refactoring - introducing generics, correcting names

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Integrate Epicyro 3.0.0

Add GFAuthConfigFactory as "alias" to AuthConfigFileFactory

Update server.policy for a new security property we use

Update authentication TCK runner to directly use epicyro factory

arjantijms committed:
Removed now unused HandlerContext, and did some cleaning

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Enable test again and add final again

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Merge branch 'master' into eleos

dmatej commented on appserver/security/appclient.security/src/main/java/com/sun/enterprise/security/appclient/ConfigXMLParser.java:
```diff
@@ -48,136 +40,134 @@
 import org.glassfish.internal.api.Globals;
 
-import static java.util.regex.Matcher.quoteReplacement;
+import com.sun.enterprise.security.common.Util;
```
Order by alphabet

dmatej commented on appserver/security/appclient.security/src/main/java/com/sun/enterprise/security/appclient/ConfigXMLParser.java:
```diff
@@ -49,22 +40,30 @@
     private static final Logger LOG = System.getLogger(ConfigXMLParser.class.getName());
 
-    private static final Pattern PROPERTY_PATTERN = Pattern.compile("\\$\\{\\{(.*?)}}|\\$\\{(.*?)}");
+    private static Pattern PROPERTY_PATTERN = Pattern.compile("\\$\\{\\{(.*?)}}|\\$\\{(.*?)}");
```
Do you reset the PROPERTY_PATTERN somewhere? If it is a constant, it should be final.

dmatej commented on appserver/security/core-ee/src/main/java/com/sun/enterprise/security/ee/JavaEESecurityLifecycle.java:
```diff
@@ -40,36 +53,69 @@ public class JavaEESecurityLifecycle implements ContainerSecurityLifecycle, Post
-        onInitialization();
+        String defaultProvidersString = null;
+        WebServicesDelegate delegate = Globals.get(WebServicesDelegate.class);
+        if (delegate != null) {
```
I would invert the condition, so it would be without negation.

dmatej commented on appserver/security/core-ee/src/main/java/com/sun/enterprise/security/jauth/ConfigFile.java:
```diff
@@ -269,11 +274,10 @@ private static ConfigParser loadParser(String className) throws IOException {
             final ClassLoader finalLoader = AuthConfig.getClassLoader();
 
-            return (ConfigParser) java.security.AccessController.doPrivileged(new java.security.PrivilegedExceptionAction() {
+            return java.security.AccessController.doPrivileged(new java.security.PrivilegedExceptionAction<ConfigParser>() {
```
```
PrivilegedExceptionAction<ConfigParser> action = () -> (ConfigParser) Class.forName(finalClassName, true, finalLoader).newInstance();
return AccessController.doPrivileged(action);
```

dmatej commented on appserver/security/core-ee/src/main/java/com/sun/enterprise/security/jmac/ConfigDomainParser.java:
```diff
@@ -221,25 +189,38 @@ private String expand(String rawProperty) {
+                } catch (IllegalStateException ee) {
+                    // log warning and give the provider a chance to
+                    // interpret value itself.
+                    _logger.log(FINE, "jmac.unexpandedproperty");
```
FINE is not a warning.

dmatej commented on appserver/security/webservices.security/src/main/java/com/sun/enterprise/security/webservices/ClientSecurityPipe.java:
```diff
@@ -16,214 +16,216 @@
+                clientAuthContext.cleanSubject(info, clientSubject);
+            }
+        } catch (Exception ex) {
+            // ignore exceptions
```
Bad practice - log the exception as debug at least, or warning.

dmatej commented on appserver/security/webservices.security/src/main/java/com/sun/enterprise/security/webservices/CommonServerSecurityPipe.java:
```diff
@@ -46,6 +47,7 @@
 /**
  * This pipe is used to do Jakarta Authentication security
  */
+@SuppressWarnings("deprecation")
```
Isn't better to see all deprecations?

arjantijms replied:
Normally yes, but if I'm not mistaken the Pipe was deprecated in favour of the Tube very shortly after the Pipe was introduced (relatively speaking), i.e. somewhere in the golden days of XML around 2005 or so.

But the Pipe has been in use more than the Tube ever since, which is close to 20 years now. It's not something that's there and should go away since it's barely used still. The Pipe is pretty much the thing you always use. It's a bit of a mess really.

I already asked to undeprecate the Pipe, since after 20 years I think we can conclude deprecating it was perhaps done in error?

cc @lukasj

arjantijms committed:
Processing feedback

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms committed:
Merge branch 'master' into eleos

arjantijms committed:
Adjust name of provider

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>

arjantijms commented on appserver/security/core-ee/src/main/java/com/sun/enterprise/security/ee/JavaEESecurityLifecycle.java:
```diff
@@ -40,36 +53,69 @@ public class JavaEESecurityLifecycle implements ContainerSecurityLifecycle, Post
+                       AuthMessagePolicy.getMessageSecurityBinding("SOAP", properties),
+                       authContextId, true);
+
+        Provider provider = new Provider("EleosProvider", "1.0", "") {
```
Good catch, thanks!

arjantijms committed:
Merge branch 'master' into eleos

arjantijms committed:
Have to keep EleosProvider for now. Update with next version of Epicyro

Signed-off-by: Arjan Tijms <arjan.tijms@omnifish.ee>
